### PR TITLE
Move form accessors into XFormInstance.objects - phase 3

### PR DESCRIPTION
**Pull request by millerdev**
Continuation of https://github.com/dimagi/commcare-hq/pull/31013 and https://github.com/dimagi/commcare-hq/pull/31019

Phase 3: Move remaining form accessors to `XFormInstance.objects`

:blowfish: Review by commit.

### Safety story

In addition to preserving all of the old tests and adapting the old methods that are being phased out to call the new method, new tests (copies of the old tests adapted to the new structure) are being added for the new model manager.

### Automated test coverage

Yes.

### QA Plan

No QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/form_processor/models/util.py [hunk 1]
```diff
@@ -14,3 +15,17 @@ def key(obj):
 
     index_map = {id_: index for index, id_ in enumerate(id_list)}
     object_list.sort(key=key)
+
+
+def attach_prefetch_models(objects_by_id, prefetched_models, link_field_name, cached_attrib_name):
```
Comment (snopoke) [inline corehq/form_processor/models/util.py:20]: should `corehq.form_processor.backends.sql.dbaccessors._attach_prefetch_models` be deleted?

File: corehq/form_processor/models/forms.py [hunk 2]
```diff
@@ -223,6 +226,25 @@ def set_archived_state(self, form, archive, user_id):
...
+            pass
+
+    @staticmethod
+    @contextmanager
+    def _unfinished_archive(form, archive, user_id, trigger_signals=True):
+        from ..change_publishers import publish_form_saved
+        with unfinished_archive(instance=form, user_id=user_id, archive=archive) as archive_stub:
+            yield archive_stub
+            is_sql = isinstance(form, XFormInstance)
+            if is_sql:
```
Comment (snopoke) [inline corehq/form_processor/models/forms.py:272]: these lines can be removed

File: corehq/form_processor/tests/test_forms.py [hunk 3]
```diff
@@ -81,6 +98,105 @@ def test_get_with_attachments(self):
         self.assertEqual('form.xml', attachment_meta.name)
         self.assertEqual('text/xml', attachment_meta.content_type)
 
+    def test_get_attachment_by_name(self):
```
Comment (snopoke) [inline corehq/form_processor/tests/test_forms.py:101]: will you delete the tests in `corehq.form_processor.tests.test_form_dbaccessor` in another PR?

**Review by snopoke — APPROVED** [review]
Small comments but nothing blocking

File: corehq/form_processor/models/util.py (hunk 1 above)
Comment (millerdev) [reply corehq/form_processor/models/util.py:20]: Yes, that will happen in a future PR as part of the case accessors refactoring.

File: corehq/form_processor/models/forms.py (hunk 2 above)
Comment (millerdev) [reply corehq/form_processor/models/forms.py:272]: Thanks for catching that. I'll do it in the next PR.

File: corehq/form_processor/tests/test_forms.py (hunk 3 above)
Comment (millerdev) [reply corehq/form_processor/tests/test_forms.py:101]: Yes, along with `FormAccessors` and `FormAccessorSQL`.

File: corehq/apps/api/object_fetch_api.py [hunk 4]
```diff
@@ -28,6 +28,7 @@
     CaseAccessors,
     get_cached_case_attachment,
 )
+from corehq.form_processor.models import XFormInstance
```
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/api/object_fetch_api.py:31]: Will resolve in a separate PR.

**Pull request merged**